src/Cm/RedisSession/Handler.php

```diff
@@ -322,35 +339,35 @@ public function __construct(ConfigInterface $config, LoggerInterface $logger, $r
                         } catch (\CredisException $e) {
                             // Prevent throwing exception if Sentinel has no password set (error messages are different between redis 5 and redis 6)
                             if ($e->getCode() !== 0 || (
-                                strpos($e->getMessage(), 'ERR Client sent AUTH, but no password is set') === false && 
+                                strpos($e->getMessage(), 'ERR Client sent AUTH, but no password is set') === false &&
                                 strpos($e->getMessage(), 'ERR AUTH <password> called without any password configured for the default user. Are you sure your configuration is correct?') === false)
                             ) {
                                 throw $e;
                             }
                         }
                     }
-                   
+
                     $sentinel = new \Credis_Sentinel($sentinelClient);
                     $sentinel
                         ->setClientTimeout($timeout)
                         ->setClientPersistent($persistent);
                     $redisMaster = $sentinel->getMasterClient($sentinelMaster);
-                    if ($pass) $redisMaster->auth($pass);
+                    if ($pass) $redisMaster->auth($pass, $username);
```

Comment:
Username is optional,  see https://redis.io/docs/latest/commands/auth/
Can you make it in code as well

Reply:
Yes, and it is currently null if config is not UsernameConfigInterface.
See here.
https://github.com/JacobBrownAustin/php-redis-session-abstract/pull/1/files#diff-fef2780a2c99c6526c10142058e117d6b01098444f4ddcee6451b4d4ca23f8f5R294

Previously, it was just using the default parameter value of null, so in case the config is not UsernameConfigInterface, then behaviour is same as previous.

Reply:
In this case, which is my new code, it is being passed to RedisCluster, 
['user' => $this->authUsername, 'pass' => $this->authUsername],
But in the extension, it only uses it if the length is non-zero.
https://github.com/phpredis/phpredis/blob/develop/redis_cluster.c#L132

Reply:
And here's where you can see the default parameter value of null for $user.
https://github.com/colinmollenhour/credis/blob/master/Client.php#L753